repos/system_upgrade/common/actors/cloud/checkhybridimage/libraries/checkhybridimage.py

```diff
@@ -39,27 +45,40 @@ def is_azure_agent_installed():
     return has_package(InstalledRPM, agent_pkg)
 
 
+def has_efi_partition():
+    """
+    Check whether ESP partition exists and is mounted.
+    """
+
+    return os.path.exists(EFI_MOUNTPOINT) and os.path.ismount(EFI_MOUNTPOINT)
+
+
 def is_bios():
-    """Check whether system is booted into BIOS"""
+    """
+    Check whether system is booted into BIOS
+    """
+
     ff = next(api.consume(FirmwareFacts), None)
     return ff and ff.firmware == 'bios'
 
 
-def check_hybrid_image():
-    """Check whether the system is using Azure hybrid image."""
-    if all([is_grubenv_symlink_to_efi(), is_azure_agent_installed(), is_bios()]):
-        api.produce(HybridImage(detected=True))
-        reporting.create_report([
-            reporting.Title(
-                'Azure hybrid (BIOS/EFI) image detected. "grubenv" symlink will be converted to a regular file'
-            ),
-            reporting.Summary(
-                'Leapp detected the system is running on Azure cloud, booted using BIOS and '
-                'the "/boot/grub2/grubenv" file is a symlink to "../efi/EFI/redhat/grubenv". In case of such a '
-                'hybrid image scenario GRUB is not able to locate "grubenv" as it is a symlink to different '
-                'partition and fails to boot. If the system needs to be run in EFI mode later, please re-create '
-                'the relative symlink again.'
-            ),
-            reporting.Severity(reporting.Severity.HIGH),
-            reporting.Groups([reporting.Groups.PUBLIC_CLOUD]),
-        ])
+def is_running_on_azure_hypervisor():
+    """
+    Check if system is running on Azure hypervisor (Hyper-V)
+    """
+
+    return detect_virt() == AZURE_HYPERVISOR_ID
+
+
+def detect_virt():
+    """
+    Detect execution in a virtualized environment
+    """
+
+    try:
+        result = run(['systemd-detect-virt'])
```

Comment:
This command is guaranteed to be installed?

Reply:
It is part of the systemd RPM